utils/error_injection.hh

```diff
@@ -163,7 +161,14 @@ class error_injection {
             if constexpr (std::is_same_v<T, std::string_view>) {
                 return s;
             } else {
-                return boost::lexical_cast<T>(s.data(), s.size());
```

Comment:
you replaced 1 line of code with 8 lines of code. I think it's a change for the worse.
Would be better if those 8 lines were extracted to a utility template function

Reply:
std::from_chars interface is quite chatty, since it tries to be more flexible than lexical_cast.

I'll see if I can improve it.